core/src/main/java/org/infinispan/AdvancedCache.java

```diff
@@ -551,4 +551,17 @@
     * @return the entry set containing all of the CacheEntries
     */
    CacheSet<CacheEntry<K, V>> cacheEntrySet();
+
+   /**
+    * Attempts to remove the entry if it is expired.  Due to expired entries not being consistent across nodes, this
+    * will still attempt to remove the value if it is not present.  Note that this will raise an expired event
+    * even if the entry is not present.  Normally this method should never be invoked except by the
+    * {@link ExpirationManager}.
```

Comment:
As an example, `ReadWriteMapImpl` and similar implementations do this very same thing, with all the components available from AdvancedCache, without the need to add any extra methods to AdvancedCache.